enterprise/kernel/src/test/java/org/neo4j/test/TestEnterpriseGraphDatabaseFactory.java

```diff
@@ -44,9 +44,37 @@
     protected GraphDatabaseBuilder.DatabaseCreator createDatabaseCreator( File storeDir,
                                                                           GraphDatabaseFactoryState state )
     {
-        return config -> {
-            config.put( "unsupported.dbms.ephemeral", "false" );
-            return new EnterpriseGraphDatabase( storeDir, config, state.databaseDependencies() );
+        return new GraphDatabaseBuilder.DatabaseCreator() {
+
+            @Override
+            public GraphDatabaseService newDatabase( Map<String,String> config )
+            {
+                config.put( "unsupported.dbms.ephemeral", "false" );
+                return new GraphDatabaseFacadeFactory( DatabaseInfo.ENTERPRISE, EnterpriseEditionModule::new )
```

Comment:
been there, tried that hehe:) That's so on a first view, it's impossible now to extract common subclass of PlatformModule because of file system problems you can't really pass that as contractor parameter since it will be used in super constructor and you will not gonna be able to customise that before usage.
It requires much bigger refactoring to do that, do not want to do that in maintenance brach